src/test/java/org/jenkinsci/plugins/parallel_test_executor/ParallelTestExecutorUnitTest.java

```diff
@@ -154,4 +159,25 @@ public void findTestOfJavaProjectDirectoryInWorkspace(){
         List<InclusionExclusionPattern> splits = ParallelTestExecutor.findTestSplits(parallelism, build, listener, true, null, new FilePath(scanner.getBasedir()), true);
         assertEquals(5, splits.size());
     }
+
+    @Test
+    public void previousBuildIsOngoing() throws IOException {
+        Job project = mock(Job.class);
+        Run previousPreviousBuild = mock(Run.class);
+        when(previousBuild.getParent()).thenReturn(project);
+        when(previousBuild.getResult()).thenReturn(null);
+        when(previousBuild.isBuilding()).thenReturn(true);
+        when(previousBuild.getPreviousBuild()).thenReturn(previousPreviousBuild);
+        when(previousBuild.getAction(eq(AbstractTestResultAction.class))).thenReturn(null);
```

Comment:
```
Unnecessary stubbings detected in test class: ParallelTestExecutorUnitTest
Clean & maintainable test code requires zero unnecessary code.
Following stubbings are unnecessary (click to navigate to relevant line of code):
  1. -> at org.jenkinsci.plugins.parallel_test_executor.ParallelTestExecutorUnitTest.previousBuildIsOngoing(ParallelTestExecutorUnitTest.java:167)
  2. -> at org.jenkinsci.plugins.parallel_test_executor.ParallelTestExecutorUnitTest.previousBuildIsOngoing(ParallelTestExecutorUnitTest.java:169)
  3. -> at org.jenkinsci.plugins.parallel_test_executor.ParallelTestExecutorUnitTest.previousBuildIsOngoing(ParallelTestExecutorUnitTest.java:171)
Please remove unnecessary stubbings or use 'lenient' strictness. More info: javadoc for UnnecessaryStubbingException class.
```

Reply:
The question in my mind is _why_ those stubbings were unnecessary. The test does not call `previousBuild.isBuilding()`?

Reply:
I think it goes first through `previousBuild.getResult()` so it doesn't execute `previousBuild.isBuilding()`